Investigate unexpected unit test failures

3 tests are going mad on Travis.
See below.
They work fine locally.

May be maven needs more memory...

`addListenerFirst_bound_singleEventOnSetValue(com.[secure].data.BinderValueChangeTest)  Time elapsed: 0.383 sec  <<< ERROR!
java.lang.RuntimeException: Internal error - please report
	at com.[secure].event.ListenerMethod.receiveEvent(ListenerMethod.java:464)
	at com.[secure].event.EventRouter.fireEvent(EventRouter.java:152)
	at com.[secure].event.EventRouter.fireEvent(EventRouter.java:115)
	at com.[secure].data.Binder.fireValueChangeEvent(Binder.java:2349)
	at com.[secure].data.Binder.access$1400(Binder.java:95)
	at com.[secure].data.Binder$BindingImpl.handleFieldValueChange(Binder.java:917)
	at com.[secure].components.data.HasValue.lambda$addValueChangeListener$3e2a9a86$1(HasValue.java:163)
	at com.[secure].flow.nodefeature.ElementPropertyMap.lambda$fireEvent$1(ElementPropertyMap.java:253)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.[secure].flow.nodefeature.ElementPropertyMap.fireEvent(ElementPropertyMap.java:253)
	at com.[secure].flow.nodefeature.ElementPropertyMap.put(ElementPropertyMap.java:106)
	at com.[secure].flow.nodefeature.AbstractPropertyMap.setProperty(AbstractPropertyMap.java:59)
	at com.[secure].flow.nodefeature.ElementPropertyMap.setProperty(ElementPropertyMap.java:64)
	at com.[secure].flow.dom.impl.BasicElementStateProvider.setProperty(BasicElementStateProvider.java:253)
	at com.[secure].flow.dom.Element.setRawProperty(Element.java:729)
	at com.[secure].flow.dom.Element.setProperty(Element.java:636)
	at com.[secure].generated.[secure].text.field.GeneratedVaadinTextField.setValue(GeneratedVaadinTextField.java:644)
	at com.[secure].data.BinderValueChangeTest.addListenerFirst_bound_singleEventOnSetValue(BinderValueChangeTest.java:159)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.IllegalAccessException: Class com.[secure].event.ListenerMethod can not access a member of class com.[secure].components.data.HasValue$ValueChangeListener with modifiers "private static"
	at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)
	at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(AccessibleObject.java:296)
	at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:288)
	at java.lang.reflect.Method.invoke(Method.java:491)
	at com.[secure].event.ListenerMethod.receiveEvent(ListenerMethod.java:459)
	... 48 more
setValue_unbound_singleEventOnSetValue(com.[secure].data.BinderValueChangeTest)  Time elapsed: 0.021 sec  <<< ERROR!
java.lang.RuntimeException: Internal error - please report
	at com.[secure].event.ListenerMethod.receiveEvent(ListenerMethod.java:464)
	at com.[secure].event.EventRouter.fireEvent(EventRouter.java:152)
	at com.[secure].event.EventRouter.fireEvent(EventRouter.java:115)
	at com.[secure].data.Binder.fireValueChangeEvent(Binder.java:2349)
	at com.[secure].data.Binder.access$1400(Binder.java:95)
	at com.[secure].data.Binder$BindingImpl.handleFieldValueChange(Binder.java:917)
	at com.[secure].components.data.HasValue.lambda$addValueChangeListener$3e2a9a86$1(HasValue.java:163)
	at com.[secure].flow.nodefeature.ElementPropertyMap.lambda$fireEvent$1(ElementPropertyMap.java:253)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.[secure].flow.nodefeature.ElementPropertyMap.fireEvent(ElementPropertyMap.java:253)
	at com.[secure].flow.nodefeature.ElementPropertyMap.put(ElementPropertyMap.java:106)
	at com.[secure].flow.nodefeature.AbstractPropertyMap.setProperty(AbstractPropertyMap.java:59)
	at com.[secure].flow.nodefeature.ElementPropertyMap.setProperty(ElementPropertyMap.java:64)
	at com.[secure].flow.dom.impl.BasicElementStateProvider.setProperty(BasicElementStateProvider.java:253)
	at com.[secure].flow.dom.Element.setRawProperty(Element.java:729)
	at com.[secure].flow.dom.Element.setProperty(Element.java:636)
	at com.[secure].generated.[secure].text.field.GeneratedVaadinTextField.setValue(GeneratedVaadinTextField.java:644)
	at com.[secure].data.BinderValueChangeTest.setValue_unbound_singleEventOnSetValue(BinderValueChangeTest.java:109)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.IllegalAccessException: Class com.[secure].event.ListenerMethod can not access a member of class com.[secure].components.data.HasValue$ValueChangeListener with modifiers "private static"
	at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)
	at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(AccessibleObject.java:296)
	at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:288)
	at java.lang.reflect.Method.invoke(Method.java:491)
	at com.[secure].event.ListenerMethod.receiveEvent(ListenerMethod.java:459)
	... 48 more
setValue_bound_singleEventOnSetValue(com.[secure].data.BinderValueChangeTest)  Time elapsed: 0.021 sec  <<< ERROR!
java.lang.RuntimeException: Internal error - please report
	at com.[secure].event.ListenerMethod.receiveEvent(ListenerMethod.java:464)
	at com.[secure].event.EventRouter.fireEvent(EventRouter.java:152)
	at com.[secure].event.EventRouter.fireEvent(EventRouter.java:115)
	at com.[secure].data.Binder.fireValueChangeEvent(Binder.java:2349)
	at com.[secure].data.Binder.access$1400(Binder.java:95)
	at com.[secure].data.Binder$BindingImpl.handleFieldValueChange(Binder.java:917)
	at com.[secure].components.data.HasValue.lambda$addValueChangeListener$3e2a9a86$1(HasValue.java:163)
	at com.[secure].flow.nodefeature.ElementPropertyMap.lambda$fireEvent$1(ElementPropertyMap.java:253)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at com.[secure].flow.nodefeature.ElementPropertyMap.fireEvent(ElementPropertyMap.java:253)
	at com.[secure].flow.nodefeature.ElementPropertyMap.put(ElementPropertyMap.java:106)
	at com.[secure].flow.nodefeature.AbstractPropertyMap.setProperty(AbstractPropertyMap.java:59)
	at com.[secure].flow.nodefeature.ElementPropertyMap.setProperty(ElementPropertyMap.java:64)
	at com.[secure].flow.dom.impl.BasicElementStateProvider.setProperty(BasicElementStateProvider.java:253)
	at com.[secure].flow.dom.Element.setRawProperty(Element.java:729)
	at com.[secure].flow.dom.Element.setProperty(Element.java:636)
	at com.[secure].generated.[secure].text.field.GeneratedVaadinTextField.setValue(GeneratedVaadinTextField.java:644)
	at com.[secure].data.BinderValueChangeTest.setValue_bound_singleEventOnSetValue(BinderValueChangeTest.java:125)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.IllegalAccessException: Class com.[secure].event.ListenerMethod can not access a member of class com.[secure].components.data.HasValue$ValueChangeListener with modifiers "private static"
	at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)
	at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(AccessibleObject.java:296)
	at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:288)
	at java.lang.reflect.Method.invoke(Method.java:491)
	at com.[secure].event.ListenerMethod.receiveEvent(ListenerMethod.java:459)
	... 48 more

`

Tests in error: 
  addListenerFirst_bound_singleEventOnSetValue(com.[secure].data.BinderValueChangeTest): Internal error - please report
  setValue_unbound_singleEventOnSetValue(com.[secure].data.BinderValueChangeTest): Internal error - please report
  setValue_bound_singleEventOnSetValue(com.[secure].data.BinderValueChangeTest): Internal error - please report

--- a/flow-components-parent/flow-components-commons/src/main/java/com/vaadin/data/Binder.java
+++ b/flow-components-parent/flow-components-commons/src/main/java/com/vaadin/data/Binder.java
@@ -2380,7 +2380,7 @@ public class Binder<BEAN> implements Serializable {
                         && !method.isDefault())
                 .collect(Collectors.toList());
         assert filteredMethods.size() == 1;
-        return methods[0];
+        return filteredMethods.get(0);
     }
 
 }

